### PR TITLE
fix vendor/k8s.io/apimachinery/pkg/api/meta staticcheck

### PR DESCRIPTION
**Pull request by habibrosyad**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Fixes `vendor/k8s.io/apimachinery/pkg/api/meta` staticcheck errors.

**Which issue(s) this PR fixes**:

Part of #92402

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/apimachinery/pkg/api/meta/meta.go [hunk 1]
```diff
@@ -40,8 +40,6 @@ func CommonAccessor(obj interface{}) (metav1.Common, error) {
 	switch t := obj.(type) {
 	case List:
 		return t, nil
-	case metav1.ListInterface:
```
Comment (lavalamp) [inline staging/src/k8s.io/apimachinery/pkg/api/meta/meta.go:43]: Why doesn't static check like this?

Comment (habibrosyad) [reply staging/src/k8s.io/apimachinery/pkg/api/meta/meta.go:43]: In https://github.com/kubernetes/kubernetes/blob/23ddc3a876c272caa8119fcf3023ab2ebd6a0035/staging/src/k8s.io/apimachinery/pkg/api/meta/interfaces.go#L33 `List` is defined as `type List metav1.ListInterface`, hence the issue with [type assertion](https://staticcheck.io/docs/checks#SA4020).

**Comment by justaugustus** [comment]
Needs rebase

**Comment by habibrosyad** [comment]
Needs /lgtm @lavalamp, thanks.

**Comment by bai** [comment]
Hey 👋 Bug Triage here. This PR has not been updated for a while, so I'd like to check on the status. The code freeze is starting 12.11.2020 (about 3-4 weeks from now) and while there is still plenty of time, we want to ensure that each PR has a chance to be merged on time.

As the PR is tagged for 1.20, is it still planned for this release?

**Comment by habibrosyad** [comment]
> Hey 👋 Bug Triage here. This PR has not been updated for a while, so I'd like to check on the status. The code freeze is starting 12.11.2020 (about 3-4 weeks from now) and while there is still plenty of time, we want to ensure that each PR has a chance to be merged on time.
> 
> As the PR is tagged for 1.20, is it still planned for this release?

@bai this needs an lgtm, which was lost because of rebase

**Comment by bai** [comment]
@lavalamp do you think you could take another look at this PR please? (Looks like you were the one who approved it previously.) 🙏

**Comment by habibrosyad** [comment]
I think I need to rebase this one more time (just in case, since last update was on Sep)

**Comment by lavalamp** [comment]
/lgtm

btw you can get an LGTM from anyone once you have the /approve, in case I miss things in my email...

**Pull request merged**